tests/test.rs

```diff
@@ -0,0 +1,105 @@
+use insta;
+use os_pipe::pipe;
+use std::io::Read;
+use std::sync::Mutex;
+use structopt::lazy_static::lazy_static;
+
+lazy_static! {
+    /// rust will try to run the tests in parallel by default, and `insta` doesn't like the
+    /// usual way of disabling this via `--test-threads=1`, so we're using this
+    /// mutex to make sure we're not re-flashing until the last run is finished
+    static ref ONE_RUN_AT_A_TIME: Mutex<i32> = Mutex::new(0i32);
+}
+
+/// run probe-run with `args` and truncate the "Finished .. in .." and "Running `...`" flashing output
+/// NOTE: this currently only capures `stdin`, so any `log::` ed output, like flashing
+fn run_and_truncate(args: &str) -> String {
+    let _guard = ONE_RUN_AT_A_TIME.lock().unwrap();
+
+    // add prefix to run this repository's version of `probe-run` and
+    // remove user-dependent registry and rustc information from backtrace paths
+    let complete_command = format!("run -- {} --shorten-paths", args);
+    let complete_command_args = complete_command.split(" ");
+
+    let (mut reader, writer) = pipe().unwrap();
+    let writer_clone = writer.try_clone().unwrap();
+
+    let mut command = std::process::Command::new("cargo");
+    command.args(complete_command_args);
+
+    // capture stderr and stdout while preserving line order
+    command.stdout(writer);
+
+    command.stderr(writer_clone);
+
+    // run `probe-run`
+    let mut handle = command.spawn().unwrap();
+
+    // force EOF
+    drop(command);
+
+    // retrieve output and clean up
+    let mut probe_run_output = String::new();
+    reader.read_to_string(&mut probe_run_output).unwrap();
+    handle.wait().unwrap();
+
+    // remove the lines printed during flashing, as they contain timing info that's not always the same
+    let mut truncated_probe_run_output = "".to_string();
+    for line in probe_run_output.lines() {
+        if !line.starts_with("    Finished")
+            && !line.starts_with("     Running `")
+            && !line.starts_with("    Blocking waiting for file lock ")
+            && !line.starts_with("   Compiling probe-run v")
+            // TODO don't drop these after https://github.com/knurling-rs/probe-run/issues/217 is resolved
+            && !line.starts_with("└─ ")
+        {
+            truncated_probe_run_output.push_str(line);
+            truncated_probe_run_output.push_str("\n");
+        }
+    }
+
+    truncated_probe_run_output
```

Comment:
```suggestion
    probe_run_output
        .lines()
        .filter(|line| {
            !line.starts_with("    Finished")
            && !line.starts_with("     Running `")
            && !line.starts_with("    Blocking waiting for file lock ")
            && !line.starts_with("   Compiling probe-run v")
            // TODO don't drop these after https://github.com/knurling-rs/probe-run/issues/217 is resolved
            && !line.starts_with("└─ ")
        })
        .map(|line| format!("{}\n", line))
        .collect()
```

Reply:
maybe it's me but I actually find that less readable=

Reply:
I personally prefer to use iterators, cause they are more compact and you need no mutable working variables. For the readability: By now I use it a lot and think it is very clear what it is doing, but maybe that is not true if you are not used to them.

🤷🏾‍♀️

Reply:
fair point about the non-mutable interim var